src/targets/gpu/mlir.cpp

```diff
@@ -623,22 +666,22 @@ struct mlir_program
     {
         auto names = m.get_parameter_names();
         std::sort(names.begin(), names.end());
-        std::vector<shape> inputs;
+        std::vector<instruction_ref> input_ins;
         std::transform(names.begin(),
                        names.end(),
-                       std::back_inserter(inputs),
-                       [&](const std::string& name) { return m.get_parameter_shape(name); });
+                       std::back_inserter(input_ins),
+                       [&](const std::string& name) { return m.get_parameter(name); });
```

Comment:
What I see here is that from the original code the `get_shape` function can do the unpacking instead of `mlir_make_shaped`, we just need to call `get_shape(m.get_parameter(name))` instead of `m.get_parameter_shape(name)` here.

Reply:
It didn't make sense to me that we have a different `get_shape` function here, so I just refactored `make_mlir_shaped`. `get_shape` is also not used for all the cases of when we get the instruction shape in this file.